### PR TITLE
Login form

### PR DESCRIPTION
**Pull request by lubegasimon**

### CODE REVIEW

File: src/validation/validation-rules.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// TODO: ref -- https://semantic-ui.com/behaviors/form.html#/usage
+$(".ui.form")
```
Comment (epicallan) [inline src/validation/validation-rules.tsx]: If we are not using this code we should get rid of it and I don't think we are. If you want to do advanced validation look into using formik

File: src/components/micro/Login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
Comment (epicallan) [inline src/components/micro/Login/index.tsx]: shouldn't be separate component

File: src/components/micro/Form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      <Button color="teal">
+        <Image spaced="right" src="https://react.semantic-ui.com/images/avatar/small/elliot.jpg" />
+        Log-in to your CSO account.
+      </Button>
+      <Segment textAlign="center" padded={true} color="grey">
+        <Form onSubmit={handleSubmit}>
+          {/* validation rules ( from ..../validation) to be applied to form using this className ( ui-validation ) */}
+          <Form.Group className="ui-validation">
+            <Form.Field>
+              <Input
```
Comment (epicallan) [inline src/components/micro/Form/index.tsx]: this should take in an email not username

File: src/components/micro/Form/index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleSubmit = (e) => {
+    props.onChange(e.state.value);
+    e.preventDefault();
+  };
+
+  return(
+    <div>
+      <Button color="teal">
+        <Image spaced="right" src="https://react.semantic-ui.com/images/avatar/small/elliot.jpg" />
```
Comment (epicallan) [inline src/components/micro/Form/index.tsx]: we don't need the avatar

File: src/components/micro/Form/index.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                icon="lock"
+                iconPosition="left"
+                type="password"
+                name="password"
+                value={form.password}
+                placeholder="Password"
+                onChange={handleChange}
+                autoFocus={props.autoFocus}
+              />
+              <Label as="a" size="medium">
```
Comment (epicallan) [inline src/components/micro/Form/index.tsx]: this should only be present on login failure from the server. And in our case when there login errors we should display contact admin to reset your password

File: src/components/micro/Form/index.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { action} from "@storybook/addon-actions";
+import {linkTo} from "@storybook/addon-links";
+import { storiesOf } from "@storybook/react";
+import React from "react";
+import LoginButton from "../Login";
+import Title from "../Title/index";
+import LoginForm from "./index";
+
+storiesOf("Form", module)
+  .addDecorator((storyFn) => <div style={{textAlign: "center"}}>{storyFn()}</div>)
```
Comment (epicallan) [inline src/components/micro/Form/index.stories.tsx]: Let's avoid these kinds of formatting decorators. Lets show the component as is.

File: src/components/micro/Form/index.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                icon="user"
+                iconPosition="left"
+                type="text"
+                name="username"
+                value={form.username}
+                placeholder="Email address or username"
+                onChange={handleChange}
+                autoFocus={props.autoFocus}
+              />
+              <br/>
```
Comment (epicallan) [inline src/components/micro/Form/index.tsx]: Avoid HTML tags for formatting. Look into using semantic UI Form groups or grid rows and columns. 
See this https://react.semantic-ui.com/collections/form/#group-variations-unstackable-group and click on the try it button

**Comment by lubegasimon** [comment]
Am opening a new PR on another branch

**Comment by epicallan** [comment]
ok but you should learn git rebase so that you don't have to close PRs with comments

**Pull request closed without merging**